assets/src/edit-story/components/inspector/documentInspector.js

```diff
@@ -1,9 +1,187 @@
+import useInspector from './useInspector';
+import { SelectMenu, InputGroup } from './shared';
+
+const ButtonCSS = css`
+	color: ${ ( { theme } ) => theme.colors.mg.v1 };
+	font-size: 14px;
+	width: 100%;
+	padding: 15px;
+	background: none;
+	margin: 5px 0px;
+`;
+const Img = styled.img`
+	width: 100%;
+	max-height: 300px
+`;
+
+const Group = styled.div`
+	border-color: ${ ( { theme } ) => theme.colors.mg.v1 };
+	display: block;
+	align-items: center;
+	margin: 15px 0px;
+`;
+
+const RemoveButton = styled.button`
+	${ ButtonCSS }
+`;
 
 function DocumentInspector() {
+	const {
+		actions: { loadStatuses, loadUsers, setDisabledStatuses },
+		state: { users, statuses, disabledStatuses },
+	} = useInspector();
+
+	const {
+		state: { meta: { isSaving }, story: { author, status, slug, date, excerpt, featuredMediaUrl, password }, capabilities },
+		actions: { updateStory, deleteStory },
+	} = useStory();
+
+	const { postThumbnails } = useConfig();
+
+	useEffect( () => {
+		loadStatuses();
+		loadUsers();
+	} );
+
+	useEffect( () => {
+		const states = ( status === 'future' ) ? [ 'pending' ] : [ 'future', 'pending' ];
+		setDisabledStatuses( states );
+	}, [ status, setDisabledStatuses ] );
+
+	const handleChangeStatus = useCallback(
+		( evt ) => 	updateStory( { properties: { status: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeAuthor = useCallback(
+		( evt ) => updateStory( { properties: { author: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangePassword = useCallback(
+		( value, evt ) => updateStory( { properties: { password: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeDate = useCallback(
+		( value, evt ) => updateStory( { properties: { date: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeExcerpt = useCallback(
+		( value, evt ) => updateStory( { properties: { excerpt: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeSlug = useCallback(
+		( value, evt ) => updateStory( { properties: { slug: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeImage = useCallback(
+		( image ) => updateStory( { properties: { featuredMedia: image.id, featuredMediaUrl: ( image.sizes && image.sizes.medium ) ? image.sizes.medium.url : image.url } } ),
+		[ updateStory ],
+	);
+
+	const handleRemoveImage = useCallback(
+		( evt ) => {
+			updateStory( { properties: { featuredMedia: 0, featuredMediaUrl: '' } } );
+			evt.preventDefault();
+		},	[ updateStory ],
+	);
+
+	const handleRemoveStory = useCallback(
+		( evt ) => {
+			deleteStory();
+			evt.preventDefault();
+		},	[ deleteStory ],
+	);
+
 	return (
-		<h2>
-			{ 'Document' }
-		</h2>
+		<>
+			<h2>
+				{ 'Document' }
+			</h2>
+			{ capabilities && capabilities.hasPublishAction && statuses && <SelectMenu
+				label="Status"
+				name="status"
+				options={ statuses }
+				disabled={ isSaving }
+				disabledOptions={ disabledStatuses }
```

Comment:
Would it be more clear to filter the statuses right away instead of passing a separate `disabledOptions`?

Reply:
Can you explain this, not sure what you mean here.

Reply:
What I meant is that perhaps instead of having a separate `disabledOptions` attribute, we could directly just filter `statuses` and pass the correct `options` right away, instead of the `SelectMenu` having to be aware of irrelevant options.

E.g. `options={ statuses.filter( status ) => ! disabledStatuses.includes( status ) }` or sth similar.

 Thoughts?

Reply:
Works for me.

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/4122/commits/c47fcf1e39e549cd797892635c1986946bedc0db